src/react/utils.js

```diff
@@ -235,6 +239,33 @@ export function forEachArrayValue(realm: Realm, array: ObjectValue, mapFunc: Fun
   }
 }
 
+export function mapArrayValue(
+  realm: Realm,
+  array: ArrayValue,
+  mapFunc: (element: Value, descriptor: Descriptor) => Value
+): ArrayValue {
+  let lengthValue = Get(realm, array, "length");
+  invariant(lengthValue instanceof NumberValue, "Invalid length on ArrayValue during reconcilation");
+  let length = lengthValue.value;
+  let newArray = Create.ArrayCreate(realm, length);
+  let returnTheNewArray = false;
+
+  for (let i = 0; i < length; i++) {
+    let elementProperty = array.properties.get("" + i);
+    let elementPropertyDescriptor = elementProperty && elementProperty.descriptor;
+    invariant(elementPropertyDescriptor, `Invalid ArrayValue[${i}] descriptor`);
```

Comment:
Array really cannot have holes?

Reply:
Not inside a ReactElement, no.

Reply:
Can you explain why in more detail? It’s not obvious to me.

Reply:
You can't in a ReactElement:

`React.createElement("div", null, , , , "div");` is not valid JS
`return React.createElement("div", null, undefined, undefined, undefined, "div");` isn't holey.